src/org/zaproxy/zap/extension/ascan/ActiveScanAPI.java

```diff
@@ -532,23 +533,41 @@ private int scanURL(URI url, User user, boolean scanChildren, boolean scanJustIn
 		StructuralNode node;
 		
 		try {
-			node = SessionStructure.find(Model.getSingleton().getSession().getSessionId(), url, method, postData);
+			Session session = Model.getSingleton().getSession();
+			node = SessionStructure.find(session.getSessionId(), url, method, postData);
 			if (node == null) {
 				throw new ApiException(ApiException.Type.URL_NOT_FOUND);
 			}
 			Target target = new Target(node);
 			target.setRecurse(scanChildren);
 			target.setInScopeOnly(scanJustInScope);
+
+			ArrayList<Object> objs = new ArrayList<Object>();
+			if (policy != null) {
+				objs.add(policy);
+			}
+
+			/*
+			 * A side effect of setting the Context based on the specified user
+			 * is the limiting of the scan to technologies associated with the
+			 * Context. When user is not specified, technologies should be
+			 * limited based on Contexts containing the supplied URL.
+			 */
```

Comment:
This is not a side effect, it is the intended effect. A user is associated to a context so when scanning as a user it's also scanning with the associated context thus constrained by its configurations (technologies, authentication, structural parameters, what's included/excluded...).